cmd/state/internal/cmdtree/internal/intercepts/cmdcall/cmdcall.go

```diff
@@ -0,0 +1,52 @@
+package cmdcall
+
+import (
+	"github.com/ActiveState/cli/internal/captain"
+	"github.com/ActiveState/cli/internal/events/cmdcall"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/pkg/project"
+)
+
+// CmdCall manages the event handling flow triggered by command calls.
+type CmdCall struct {
+	primer *primer.Values
+}
+
+// New returns a pointer to a prepared CmdCall instance.
+func New(p *primer.Values) *CmdCall {
+	return &CmdCall{
+		primer: p,
+	}
+}
+
+// InterceptExec handles the before command logic, calls the next ExecuteFunc,
+// and then handles the after command logic.
+func (c *CmdCall) InterceptExec(next captain.ExecuteFunc) captain.ExecuteFunc {
+	return func(cmd *captain.Command, args []string) error {
+		cc := cmdcall.New(c.primer, cmd.UseFull())
```

Comment:
What's with the recursion? We are already in a CmdCall.

Reply:
Same as other concern. https://github.com/ActiveState/cli/pull/1052#discussion_r515408992

Reply:
I'm not sure I understand why the location of the package, which is addressed in the comment you linked, is related to my concern here?

Reply:
The cmdcall.New is setting up a command call event handler, and the cmdcall struct in this package is the command call intercept context.